include/nuttx/can/can_sender.h

```diff
@@ -0,0 +1,236 @@
+/****************************************************************************
+ * include/nuttx/can/can_sender.h
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_CAN_SENDER_H
+#define __INCLUDE_NUTTX_CAN_SENDER_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/compiler.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <nuttx/list.h>
+#include <nuttx/fs/fs.h>
+#include <nuttx/fs/ioctl.h>
+#include <nuttx/mutex.h>
+#include <nuttx/can/can.h>
+
+#ifdef CONFIG_CAN
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_CAN_TXPRIORITY 
+
+/* There are three linked lists to manage TX buffer:
+ * tx_send  - cache message send from application but not send to H/W.
+ *            Sorted in ascending order based on can_id.
+ * tx_untcf - cache message send to H/W but not confirmation.  Sorted in
+ *            ascending order based on can_id.
+ * tx_free  - can_write function get a tx_free node, write message, then move
+ *            this node to tx_send list.
+ */
+
+  #define FREE_FULL(sender) \
```

Comment:
This is GCC extension and is C89 incompatible

Reply:
Done